mne/decoding/xdawn.py

```diff
@@ -0,0 +1,177 @@
+"""Contains refactored Xdawn class."""
+
+# Authors: Alexandre Barachant <alexandre.barachant@gmail.com>
+#          Asish Panda <asishrocks95@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+
+from ..preprocessing.xdawn import _Xdawn
+from . import TransformerMixin
+
+
+class XdawnTransformer(TransformerMixin, _Xdawn):
+
+    """Implementation of the Xdawn Algorithm.
+
+    Xdawn is a spatial filtering method designed to improve the signal
+    to signal + noise ratio (SSNR) of the ERP responses. Xdawn was originally
+    designed for P300 evoked potential by enhancing the target response with
+    respect to the non-target response. This implementation is a light version
+    of the original version and follows scikit-learn API strictly.
+
+    .. note:: This does not handle overlapping events. Use original
+              preprocessing.xdawn instead.
+
+    Parameters
+    ----------
+    n_components : int (default 2)
+        The number of components to decompose M/EEG signals.
+    signal_cov : None | Covariance | ndarray, shape (n_channels, n_channels)
+        (default None). The signal covariance used for whitening of the data.
+        if None, the covariance is estimated from the epochs signal.
+    reg : float | str | None (default None)
+        if not None, allow regularization for covariance estimation
+        if float, shrinkage covariance is used (0 <= shrinkage <= 1).
+        if str, optimal shrinkage using Ledoit-Wolf Shrinkage ('ledoit_wolf')
+        or Oracle Approximating Shrinkage ('oas').
+
+    Attributes
+    ----------
+    filters_ : dict of ndarray
+        The Xdawn components used to decompose the data for each event type.
+    patterns_ : dict of ndarray
+        The Xdawn patterns used to restore M/EEG signals for each event type.
+
+
+    See Also
+    --------
+    mne.preprocessing.Xdawn
+    CSP
+
+    References
+    ----------
+    [1] Rivet, B., Souloumiac, A., Attina, V., & Gibert, G. (2009). xDAWN
+    algorithm to enhance evoked potentials: application to brain-computer
+    interface. Biomedical Engineering, IEEE Transactions on, 56(8), 2035-2043.
+
+    [2] Rivet, B., Cecotti, H., Souloumiac, A., Maby, E., & Mattout, J. (2011,
+    August). Theoretical analysis of xDAWN algorithm: application to an
+    efficient sensor selection in a P300 BCI. In Signal Processing Conference,
+    2011 19th European (pp. 1382-1386). IEEE.
+    """
+
+    def __init__(self, n_components=2, signal_cov=None, reg=None):
+        """init xdawn."""
+        # XXX correct_overlap is not yet supported
+        # This attribute is introduced to make it compatible with parent
+        # class
```

Comment:
what will we do with these XXX? if it is an informative comment for future readers remove XXX. otherwise we will think that it prevents merging